src/hooks/useOverlay.tsx

```diff
@@ -0,0 +1,40 @@
+import {useOnOutsideClick, useOpenAndCloseFocus, useOnEscapePress} from './index'
+import {TouchOrMouseEvent} from './useOnOutsideClick'
+import {AnchoredPositionHookSettings, useAnchoredPosition} from './useAnchoredPosition'
+import {useRef} from 'react'
+
+export type UseOverlaySettings = {
+  ignoreClickRefs: React.RefObject<HTMLElement>[]
```

Comment:
This should be optional, to align with `useOnOutsideClick`: https://github.com/primer/components/blob/46c6b804ea38d29f58a4c1a92238a6a6c3729667/src/hooks/useOnOutsideClick.tsx#L7

Reply:
nice find 🚀